File: cpp/2974_number_game/number_game.cpp
#include <cassert>
#include <tuple>
#include <vector>

using namespace std;

class Solution {
 public:
  vector<int> numberGame(vector<int>& nums) {
    sort(nums.begin(), nums.end());
    for (int i = 0; i < nums.size(); i += 2) {
      swap(nums[i], nums[i + 1]);
    }
    return nums;
  }
};

int main() {
  vector<tuple<vector<int>, vector<int>>> tests{
      {{5, 4, 2, 3}, {3, 2, 5, 4}},
      {{2, 5}, {5, 2}},
  };

  for (auto& [nums, ans] : tests) {
    assert(Solution().numberGame(nums) == ans);
  }
}

File: rust/2974_number_game/number_game.rs
struct Solution;

impl Solution {
    pub fn number_game(mut nums: Vec<i32>) -> Vec<i32> {
        nums.sort();
        for i in (0..nums.len() - 1).step_by(2) {
            nums.swap(i, i + 1);
        }
        nums
    }
}

fn main() {
    let tests = vec![
        (vec![5, 4, 2, 3], vec![3, 2, 5, 4]),
        (vec![2, 5], vec![5, 2]),
    ];

    for (nums, ans) in tests {
        assert_eq!(Solution::number_game(nums), ans);
    }
}
